src/WebJobs.Script/Binding/ExtensionBinding.cs

```diff
@@ -54,6 +54,10 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes(Type para
             {
                 await BindAsyncCollectorAsync<JObject>(context);
             }
+            else if (_binding.DefaultType == typeof(IAsyncCollector<string>))
+            {
+                await BindAsyncCollectorAsync<string>(context);
+            }
             else if (_binding.DefaultType == typeof(Stream))
             {
```

Comment:
I think we should have this method throw an exception if it gets a `DefaultType` that it doesn't know how to handle. That would have caught this right away. Since we don't have a limitation on this, people could be returning all kinds of things that we'd no-op on, which doesn't seem like the behavior we want.